docs/index.asciidoc

```diff
@@ -55,17 +55,26 @@ the new template is installed.
 
 ==== Retry Policy
 
-The retry policy has changed significantly in the 2.2.0 release.
+The retry policy has changed significantly in the 8.0.3 release.
 This plugin uses the Elasticsearch bulk API to optimize its imports into Elasticsearch. These requests may experience
-either partial or total failures.
+either partial or total failures. The bulk API sends batches of requests to an HTTP endpoint. Error codes are returned for the
+batch as a whole and for individual documents
 
-The following errors are retried infinitely:
+The following HTTP request errors to the bulk API are retried infinitely:
 
-- Network errors (inability to connect)
-- 429 (Too many requests) and
-- 503 (Service unavailable) errors
+- Network errors (Inability to connect)
+- 404 (Not found) Indicating an upstream proxy has no matching route
+- 429 (Too many requests) Graceful backoff indication by Elastic
+- 500 (Internal Server Error) when an ES node disconnect occurs this is temporarily returned
+- 502 (Bad gateway) During shutdown a proxy might return this
+- 503 (Service unavailable) Proxy returns an unavailable service
+- 504 (Gateway timed out) Proxy backend timed out
 
-NOTE: 409 exceptions are no longer retried. Please set a higher `retry_on_conflict` value if you experience 409 exceptions.
+The following document errors are handled as follows:
+- 400 and 404 errors are sent to the DLQ (if enabled)
+- 409 errors (conflict) are logged as a warn and dropped. If you'd like these retried
```

Comment:
@jordansissel apparently my brain drops sentences after a certain point, I need to work on its retry logic too! :)